pysph/sph/surface_tension.py

```diff
@@ -500,72 +821,230 @@ def loop(self, d_idx, s_idx, d_V, s_V, d_rho, s_rho,
                   d_nx, d_ny, d_nz, d_ddelta, d_N):
         # absolute value of the color gradient
         mod_gradc2 = d_cx[d_idx]*d_cx[d_idx] + \
             d_cy[d_idx]*d_cy[d_idx] + \
             d_cz[d_idx]*d_cz[d_idx]
 
         # avoid sqrt computations on non-interface particles
-        if mod_gradc2 > 1e-6:
+        h2 = d_h[d_idx]*d_h[d_idx]
+        if mod_gradc2 > 1e-4/h2:
             # this normal is reliable in the sense of [JM00]
             d_N[d_idx] = 1.0
 
             # compute the normals
-            mod_gradc = 1./sqrt(mod_gradc2)
+            one_mod_gradc = 1./sqrt(mod_gradc2)
 
-            d_nx[d_idx] = d_cx[d_idx] * mod_gradc
-            d_ny[d_idx] = d_cy[d_idx] * mod_gradc
-            d_nz[d_idx] = d_cz[d_idx] * mod_gradc
+            d_nx[d_idx] = d_cx[d_idx] * one_mod_gradc
+            d_ny[d_idx] = d_cy[d_idx] * one_mod_gradc
+            d_nz[d_idx] = d_cz[d_idx] * one_mod_gradc
 
             # discretized dirac delta
-            d_ddelta[d_idx] = 1./mod_gradc
-
-# FIXME: The implementation based on the formulation presented in
-# [A10] seems to be incorrect.
+            d_ddelta[d_idx] = 1./one_mod_gradc
 
 
-class AdamiReproducingDivergence(Equation):
-    r"""Reproducing divergence approximation Eq. (20) in [A10] to
-    compute the curvature
-
-    .. math::
-
-        \nabla \cdot \boldsymbol{\phi}_a = d\frac{\sum_b
-        \boldsymbol{\phi}_{ab}\cdot \nabla_a
-        W_{ab}V_b}{\sum_b\boldsymbol{x}_{ab}\cdot \nabla_a W_{ab} V_b}
-
+def get_surface_tension_equations(fluids, solids, scheme, rho0, p0, c0, b,
+                                  factor1, factor2, nu, sigma, d, epsilon,
+                                  gamma, real=False):
     """
-    def __init__(self, dest, sources, dim):
-        self.dim = dim
-        super(AdamiReproducingDivergence, self).__init__(dest, sources)
-
-    def initialize(self, d_idx, d_kappa, d_wij_sum):
-        d_kappa[d_idx] = 0.0
-        d_wij_sum[d_idx] = 0.0
-
-    def loop(self, d_idx, s_idx, d_kappa, d_wij_sum,
-             d_nx, d_ny, d_nz, s_nx, s_ny, s_nz, d_V, s_V,
-             DWIJ, XIJ, RIJ, EPS):
-        # particle volumes
-        Vi = 1./d_V[d_idx]
-        Vj = 1./s_V[s_idx]
-
-        # dot product in the numerator of Eq. (20)
-        nijdotdwij = (d_nx[d_idx] - s_nx[s_idx]) * DWIJ[0] + \
-            (d_ny[d_idx] - s_ny[s_idx]) * DWIJ[1] + \
-            (d_nz[d_idx] - s_nz[s_idx]) * DWIJ[2]
-
-        # dot product in the denominator of Eq. (20)
-        xijdotdwij = XIJ[0]*DWIJ[0] + XIJ[1]*DWIJ[1] + XIJ[2]*DWIJ[2]
-        xijdotdwij /= (RIJ + EPS)
-
-        # accumulate the contributions
-        d_kappa[d_idx] += nijdotdwij * Vj
-        d_wij_sum[d_idx] += RIJ * xijdotdwij * Vj
-
-    def post_loop(self, d_idx, d_kappa, d_wij_sum):
-        # normalize the curvature estimate
-        d_kappa[d_idx] /= d_wij_sum[d_idx]
-        d_kappa[d_idx] *= self.dim
+    This function returns the required equations for the multiphase
```

Comment:
Can you please elaborate the parameters, supported schemes.  Also describe the parameters properly, for example like this: https://github.com/pypr/pysph/blob/master/pysph/sph/scheme.py#L885